src/saving/serializers/ThriveJsonConverter.cs

```diff
@@ -347,23 +347,34 @@ public static bool IsIgnoredGodotMember(string name, Type type)
         // Detect ref to already loaded object
         var refId = item[REF_PROPERTY];
 
-        if (refId != null)
+        if (refId is { Type: JTokenType.String })
         {
-            return serializer.ReferenceResolver.ResolveReference(serializer, refId.Value<string>());
+            var refValue = refId.Value<string>();
+
+            if (refValue == null)
+                throw new JsonException($"{REF_PROPERTY} read as string failed");
```

Comment:
This very code is repeated 3 times with a single variable name, so we might want to factor it?

Reply:
Well I considered suppressing the warning because Value<string> should *not* be able to ever return null after there is that check that the token type is a string. I thought it would be better like this.

Reply:
Hmmm... I guess it would be a bit non-sensical to factor just to add a throwing exception that ismostly here to appease the checks indeed. That's a sufficient argument for me to accept this design.

Reply:
I already rewrote this as an extension method (though without the per-key specialized error message, but that shouldn't matter as I said, this condition should never be reached).

Reply:
> Value should not be able to ever return null after there is that check that the token type is a string

Perhaps it could be worth to mention this in a comment though?

Reply:
I mean there are `.Value` calls over the codebase, I don't feel like explaining that in a comment here that that conversion succeeds as long as you have checked the json token type beforehand, would help as this is like a general JSON library usage thing.

Reply:
Sounds a fair argument to me.